src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,114 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] listOfElements;
+    private int size;
+
+    ArrayList() {
+        listOfElements = (T[]) new Object[DEFAULT_CAPACITY];
+        size = 0;
+    }
+
     @Override
     public void add(T value) {
-
+        if (size < listOfElements.length) {
+            listOfElements[size++] = value;
+        } else {
+            grow(size + 1);
+            listOfElements[size++] = value;
+        }
     }
 
     @Override
     public void add(T value, int index) {
-
+        checkLengthForAdd(index);
```

Comment:
let`s use checkIndex() method instead of checkLengthForAdd, also you need to check, if index = size you need to use add()